ckan/lib/helpers.py

```diff
@@ -383,6 +384,14 @@ def url_for(*args, **kw):
     return _local_url(my_url, locale=locale, **kw)
 
 
+def _coerce_for_querystring(value):
+    if value is None:
+        return 'null'
+    if isinstance(value, numbers.Number):
+        return str(value)
```

Comment:
To keep with the JSON style `null`:
```suggestion
        return str(value).lower()  # render True/False as true/false
```

Reply:
Having churned this over slightly more over the weekend, it seems to me that really in this fallback case `except FlaskRouteBuildError:` we should really be trying to match the behaviour of named routes as closely as possible.

i.e: what we should be aiming for is that

`h.url_for("dataset.read", id="my_dataset", somebool=True)` and
`h.url_for("/dataset/my_dataset", somebool=True)`

should do the same thing, as should

`h.url_for("dataset.read", id="my_dataset", empty=None)` and
`h.url_for("/dataset/my_dataset", empty=None)`

etc


FWIW:
```
h.url_for("dataset.read", id="my_dataset", param=True) == '/dataset/my_dataset?param=True'
h.url_for("dataset.read", id="my_dataset", param=None) == '/dataset/my_dataset'
h.url_for("dataset.read", id="my_dataset", param={}) == '/dataset/my_dataset?param=%7B%7D'
```

Shall I alter it so the fallback matches the named route behaviour? @amercader @wardi ? It seems the most logical on reflection.

Reply:
Yes, @chris48s this makes sense. It's good to have `url_for` having a consistent output regardless of how it is invoked

Reply:
OK. I've updated to implement that behaviour